testng-core-api/src/main/java/org/testng/ITestClassInstance.java

```diff
@@ -0,0 +1,36 @@
+package org.testng;
+
+/** Represents the ability to retrieve the parameters associated with a factory method. */
+public interface ITestClassInstance {
+
+  /** @return - The actual instance associated with a factory method */
+  Object getInstance();
```

Comment:
Could you try to make it generic?

Reply:
Generic as in `T getInstance()` ? I wouldn't know the type of `T` anywhere because entire TestNG works with the type of test class to be `Object` no ?

Reply:
It's just for the beauty because instance will always considered as <?>

Reply:
Ok. But when there's no functional value, I am not sure as to what would the beautification do here.

Reply:
No functional value before we need another generic input/output. It is just a kind of documentation without any cost.